lib/bson/bson.js

```diff
@@ -93,15 +104,14 @@ BSON.prototype.serializeWithBufferAndIndex = function(object, finalBuffer, optio
 
   // Attempt to serialize
   var serializationIndex = serializer(
-    buffer,
+    finalBuffer,
```

Comment:
this looks like an unrelated change?

Reply:
yeah, that's an optimization. it saves one copy since you can just flush to the final buffer instead of flushing to a temp buffer and then that to the final buffer

Reply:
that's fine, but if you don't mind I'd prefer it in another PR so we can clearly call it out in the changelog